server/src/oauth.ts

```diff
@@ -0,0 +1,83 @@
+import OAuth2Strategy from 'passport-oauth2'
+import axios from 'axios'
+import { Role, User } from './entity/User'
+import env from 'dotenv-safe'
+import path from 'path'
+
+env.config({
+  path: path.resolve(process.cwd(), '..', '.env'),
+  example: path.resolve(process.cwd(), '..', '.env.example')
+})
+
+if (process.env.CLIENT_ID === undefined || process.env.CLIENT_SECRET === undefined) {
+  throw new Error('CLIENT_ID AND/OR CLIENT_SECRET ARE NOT SET')
+}
+
+export const oauthStrategy = new OAuth2Strategy({
+  authorizationURL: 'https://gemont.de/iserv/oauth/v2/auth',
+  tokenURL: 'https://gemont.de/iserv/oauth/v2/token',
+  clientID: process.env.CLIENT_ID!,
+  clientSecret: process.env.CLIENT_SECRET!,
+  callbackURL: 'https://localhost:4000/api/callback',
+  scope: ['profile', 'email', 'openid', 'roles', 'groups', 'uuid']
```

Comment:
Was brauchen wir hiervon tatsächlich? Kann _email_ z.B. weg?

Reply:
Zur Bewertung mal ein Screenshot:

![Student Return OAuth](https://user-images.githubusercontent.com/77128595/144288064-958cf4e2-182e-4a5c-ad6e-785ee434ae61.jpg)

Für die Entwicklung war es gut, alles drin zu haben, aber jetzt könnten mMn raus: email, groups, uuid. Openid weiß ich gar nicht, was das ist... Wir verwenden nur Namen und Rollen.

Die Iserv-Admins können ebenfalls festlegen, was wir bekommen. Das nur als Anmerkung, ich denke nicht, dass das deckungsgleich sein muss.

Reply:
Probably we shouldn't be to quick deleting them (#129)